nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -2347,11 +2329,16 @@ private boolean mayBeNullExpr(VisitorState state, ExpressionTree expr) {
     return exprMayBeNull && nullnessFromDataflow(state, expr);
   }
 
-  private boolean mayBeNullMethodCall(Symbol.MethodSymbol exprSymbol) {
+  private boolean mayBeNullMethodCall(
+      Symbol.MethodSymbol exprSymbol, MethodInvocationTree invocationTree, VisitorState state) {
     if (codeAnnotationInfo.isSymbolUnannotated(exprSymbol, config)) {
       return false;
     }
-    return Nullness.hasNullableAnnotation(exprSymbol, config);
+    return Nullness.hasNullableAnnotation(exprSymbol, config)
+        || (config.isJSpecifyMode()
```

Comment:
Same here, let's break this into a bunch of separate checks. It's more verbose, but easier to read (and to see at a glance what is gated by JSpecify mode 🙂